apps/src/lab2/levelEditors/predictSettings/EditPredictSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useState} from 'react';
+import {LevelPredictSettings, PredictQuestionType} from '../types';
+import {SimpleDropdown} from '@cdo/apps/componentLibrary/dropdown';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+import moduleStyles from './edit-predict-settings.module.scss';
+import {BodyThreeText} from '@cdo/apps/componentLibrary/typography';
+import FreeResponseFields from './FreeResponseFields';
+import MultipleChoiceFields from './MultipleChoiceFields';
+
+interface EditPredictSettingsProps {
+  initialSettings: LevelPredictSettings;
+}
+
+const EditPredictSettings: React.FunctionComponent<
+  EditPredictSettingsProps
+> = ({initialSettings}) => {
+  const [predictSettings, setPredictSettings] = useState<LevelPredictSettings>(
+    initialSettings || {is_predict_level: false}
+  );
+
+  const handleIsPredictToggle = () => {
+    setPredictSettings({
+      ...predictSettings,
+      is_predict_level: !predictSettings.is_predict_level,
+      question_type:
+        predictSettings.question_type || PredictQuestionType.FreeResponse,
+    });
+  };
+
+  const handleQuestionTypeChange = (
+    e: React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    setPredictSettings({
+      ...predictSettings,
+      question_type: e.target.value as PredictQuestionType,
+    });
+  };
+
+  const renderMultipleChoiceOptions = () => {
+    if (!predictSettings.multiple_choice_options) {
+      setPredictSettings({
+        ...predictSettings,
+        multiple_choice_options: [''],
```

Comment:
Could this be set in the initial useState assignment so it doesn't have to be checked every time?

Reply:
good call